### DISCUSSION TITLE
Authenticating to GCP Artifact registry

### DISCUSSION THREAD

**Discussion by dogmatic69** [renovatebot/renovate#23124] [Request Help]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I am trying to get this working for the first time

### Describe the problem

Unable to configure hostRules to authenticate to GCP Artifact registry. I've tried every manner of config and trawled the source code looking for clues.

The only way I've managed to get it working was to set npmrc with an encrypted copy of the whole `.npmrc` (which means the encryption and permissions are correct)

npmrc config is directly from `gcloud artifacts print-settings npm --repository npm --location europe-west3 --scope @redacted`

looks like this: (after removing some fluff)
```
@redacted:registry=https://europe-west3-npm.pkg.dev/xxxx/npm/
//europe-west3-npm.pkg.dev/xxxx/npm/:_password="redacted"
//europe-west3-npm.pkg.dev/xxxx/npm/:username=_json_key_base64
```

Before using "npmrc" config, I've attempted to use the password from the generated config in the host rules with username/password etc. Encrypted as is or with one or more base64 encodes etc.

I figured it should be pretty straight forward but have not been able to find the right combination as per docs online.

### Relevant debug logs

<details><summary>Logs</summary>
one example of not working config.

```
DEBUG: Repository config
{
  "fileName": "renovate.json"
  "config": {
    "extends": [
      "config:base"
    ],
    ...
    "hostRules": [
      {
        "hostType": "npm",
        "matchHost": "https://europe-west3-npm.pkg.dev/xxxx/npm/",
        "username": "_json_key_base64",
        "encrypted": {
          "password": "***********"
        }
      }
    ]
  }
}

```

example of working config

```
DEBUG: Repository config
{
  "fileName": "renovate.json"
  "config": {
    "extends": [
      "config:base"
    ],
    ...
    "encrypted": {
      "npmrc": "***********"
    },
    "hostRules": [
    ]
  }
}

```

</details>

### Have you created a minimal reproduction repository?

I have explained in the description why a minimal reproduction is impossible

**Comment by viceice** [comment]
you need to decode the base64 once from config before encrypt
